### PR TITLE
Bug 1172155 - Show a warning when switching to UMS on devices with a limited support r=eragonj

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1172155

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=77053eaa46af3f7d8f790fa8b391ae470fe72eaf

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=98e3f483c23fe90d26b81b90fd56057b89d783e6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3331256c285a5a219715e0ac55cc4797493243fa

File: apps/settings/js/modules/panel_utils.js [hunk 1]
```diff
@@ -382,35 +382,41 @@ define(function(require) {
...
+          case 'select':
+            for (i = 0, count = input.options.length; i < count; i++) {
+              if (input.options[i].value == value) {
+                input.options[i].selected = true;
+                break;
+              }
+            }
+            break;
+          case 'radio':
+            input.checked = (input.value == value);
```
Comment (EragonJ) [inline apps/settings/js/modules/panel_utils.js]: `===`

File: apps/settings/js/modules/panel_utils.js [hunk 2]
```diff
@@ -382,35 +382,41 @@ define(function(require) {
...
+            break;
+          case 'range':
+            if (input.value == value) {
+              return;
+            }
+            input.value = value;
+            break;
+          case 'select':
+            for (i = 0, count = input.options.length; i < count; i++) {
+              if (input.options[i].value == value) {
```
Comment (EragonJ) [inline apps/settings/js/modules/panel_utils.js]: `===`

File: apps/settings/js/modules/panel_utils.js [hunk 3]
```diff
@@ -382,35 +382,41 @@ define(function(require) {
...
+            if (input.checked == value) {
+              return;
             }
-          }
-          break;
-      }
+            input.checked = value;
+            break;
+          case 'range':
+            if (input.value == value) {
```
Comment (EragonJ) [inline apps/settings/js/modules/panel_utils.js]: `===`

File: apps/settings/js/modules/panel_utils.js (hunk 1 above)
Comment (crh0716) [inline apps/settings/js/modules/panel_utils.js]: `value` could be undefined so we need a conversion. I turn to use ` === !!value`.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=056c2bddda91c0f46e4a7193bb324d5a9d06f902

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=767d1a60318632496b959e51d45a1ee7c60ce77d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=97a8f92fd019ee65902b73053c3f08456f6906ca

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=110826f29d27cfe483eab8f01e9853d03a7f395d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b0c8d6579278ec456e9534754f36d2c6eaf796b2

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=361614f820dfc32f1299469a02df9dc4044208e9

**Pull request merged**